### ISSUE TITLE
factories not auto cleaned up with db transactions

### ISSUE THREAD

**Issue by bradfordcondon** [statonlab/TripalTestSuite#30]
## Problem:
MY DB is flooded with fake CVs and organisms after using factories.

## To reproduce:
In my ctest class, I invoke transactions like so:

```
<?php

use StatonLab\TripalTestSuite\TripalTestCase;
use StatonLab\TripalTestSuite\DBTransaction;

class entity_child_cvterm_test extends TripalTestCase {

  use DBTransaction;

  static $obj;


```

In my test, I have:

```    
$organism = factory('chado.organism')->create();
     $features =  factory('chado.feature', 100)->create();

```

I don't specify anything in setUp or breakdown but i dont think I need to?

**Comment by almasaeed2010** [comment]
This is odd. I added new tests that check if the data still exist after the transaction has ended and it succeeded, which means it is behaving as expected. 
Can you share a link to the module that has this issue so I can attempt to replicate?

**Comment by bradfordcondon** [comment]
https://github.com/statonlab/tripal_cv_xray

**Comment by almasaeed2010** [comment]
I still can't replicate this issue. 

Here is the method I used to test this. I added the following code to tripal_cv_xray tests:
```
  use DBTransaction;

  public function test_using_factories() {
    $obj = $this->obj;

    factory('chado.organism', 100);
    factory('chado.feature', 100);
 }
```

Then:
1. Before executing the tests, I ran a count query for organisms in the db (had 4) and for features (586).
1. Ran the tests
1. Reran the count queries and got the same results.

The issue might be just that DBTransaction trait wasn't present during one or more the phpunit runs.

Let me know if you encounter the issue again.
